core/services/routing/RoutingSwitch.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+
+namespace EventEspresso\core\services\routing;
+
+
+use EventEspresso\core\exceptions\ExceptionStackTraceDisplay;
+use EventEspresso\core\services\loaders\LoaderInterface;
+use Exception;
+
+/**
+ * Class RoutingSwitch
+ *
+ * @package EventEspresso\core\services\routing
+ * @since $VID:$
+
+ */
+class RoutingSwitch
+{
+
+    /**
+     * @var LoaderInterface $loader
+     */
+    private $loader;
+
+    /**
+     * @var RouteHandler $router
+     */
+    private $router;
+
+
+    /**
+     * RoutingSwitch constructor.
+     *
+     * @param LoaderInterface $loader
+     * @param RouteHandler    $router
+     */
+    public function __construct(LoaderInterface $loader, RouteHandler $router)
+    {
+        $this->loader = $loader;
+        $this->router = $router;
+    }
+
+
+    /**
+     * @throws Exception
+     */
+    public function handle()
+    {
+        try {
+            // now load and prep all primary Routes
+            $this->router->addRoute('EventEspresso\core\domain\entities\routing\handlers\admin\ActivationRequests');
+            $this->router->addRoute('EventEspresso\core\domain\entities\routing\handlers\shared\RegularRequests');
+            $route_request_type = $this->router->getRouteRequestType();
+            switch($route_request_type) {
+                case PrimaryRoute::ROUTE_REQUEST_TYPE_ACTIVATION:
+                    $this->loadActivationRoutes();
+                    break;
+                case PrimaryRoute::ROUTE_REQUEST_TYPE_REGULAR:
+                    $this->loadRegularRoutes();
+                    break;
```

Comment:
moved all of the route loading out of EE_System into this new class because I didn't want to pepper EE_System with all kinds of conditionals.
This has to utilize hooks unfortunately, so not as direct as before, but a little more clean, DRY , and SRP friendly.